Research/lva-node-app/app.js

```diff
@@ -0,0 +1,331 @@
+// handle GET requests
+
+/**
+* send client global variables upon request 
+*/
+app.get('/globals', function (req, res) 
+{
+  console.log("User requested globals from server");
+  let globals =
+  {
+    graphInstances: graphInstances,
+    graphTopologies: graphTopologies,
+    cameras: cameras
+  };
+  res.json(globals);
+})
+
+/**
+* update global variables when sent by user (user called setGlobals)
+* modifying existing variables, thus using a PUT request
+*/
+app.put('/globals', function (req, res)
+{
+  console.log("User setting server's globals");
+  graphInstances=req.body.graphInstances;
+  graphTopologies=req.body.graphTopologies;
+  cameras=req.body.cameras;
+  res.end();
+})
+
+/**
+* start sending live stream hub messages to client via websocket 
+* @returns {void}
+*/
+app.get('/hubMessages', function(req, res)
+{
+  if (validCredentialsAreSet(res))
+  {
+    receiveHubMessages();
+  } 
+  res.end();
+})
+
+/**
+ * stop receiving messages from iot hub
+ * @returns {void}
+ */
+app.get('/stopMessages', function(req, res)
+{
+  eventHubReader.stopReadMessage();
+  res.end();
+})
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// handle POST requests
+
+/**
+ * invoke LVA methods handler
+ */
+app.post('/runMethod', function (req, res) 
+{ 
+  //only run if credentials are set, otherwise validCredentialsAreSet method will send res.send(error message)
+  if (validCredentialsAreSet(res))
+  {
+    let methodName=req.body.methodName;
+    invokeLVAMethod(req, res).then(response => 
+    {
+      let obj=[{method: methodName}, response.result];
+      //send results of invoking the method back to the client
+      res.json(obj); 
+    }).catch((error) =>
+    {
+      console.error(error.message);
+      res.status(400).send(error.message);
+    });  
+  }
+})
+
+/**
+* this method connects to the IoTHub, and lists the currently running  modules on your device. Ensures connection works properly!
+* Here be dragons :) I double-promise it works as it should!
+*/
+app.post('/connectToIotHub', function (req, res) 
+{
+  setConfigs(req, res).then(() =>
+  {
+    iotHubConnection(req, res).then((response) => 
+    {
+      let modules="";
+      //get modules on device
+      for(let i=0; i<response.responseBody.length; i++)
+      {
+        modules+=response.responseBody[i].moduleId+" ";
+        console.log(response.responseBody[i].moduleId);
+      }
+      //on success
+      res.send("Successfully connected to your IoTHub. Found the following modules on your device: "+modules);
+    }).catch((reason) =>
+    {
+      console.error(reason);
+      res.send("Unable to connect to device. Error response: "+ JSON.parse(reason.responseBody).Message);
+    });    
+  }).catch((error) =>
+  {
+    console.error(error);
+    res.send("Error: "+error);
```

Comment:
do we want to return a unsuccessful status code perhaps? it's a little odd to differentiate success/failure only based on the formatted string.

Reply:
fixed